lib/bundler/source/git/git_proxy.rb

```diff
@@ -136,15 +136,28 @@ def has_revision_cached?
         end
 
         # Escape the URI for git commands
-        def uri_escaped
+        def uri_escaped_with_configured_credentials
+          remote = configured_uri_for(uri)
           if Bundler::WINDOWS
             # Windows quoting requires double quotes only, with double quotes
             # inside the string escaped by being doubled.
-            '"' + uri.gsub('"') { '""' } + '"'
+            '"' + remote.gsub('"') {|s| '""'} + '"'
           else
             # Bash requires single quoted strings, with the single quotes escaped
             # by ending the string, escaping the quote, and restarting the string.
-            "'" + uri.gsub("'") { "'\\''" } + "'"
+            "'" + remote.gsub("'") {|s| "'\\''"} + "'"
+          end
+        end
+
+        # Adds credentials to the URI as Fetcher#configured_uri_for does
+        def configured_uri_for(uri)
+          if /https?:/ =~ uri
```

Comment:
I copied this code from `Fetcher#configured_uri_for`.
And more over, `URI.regexp =~ "git@github.com:bundler/bundler.git"` returns truthy but `URI("git@github.com:bundler/bundler.git")` raises exception:

```
irb(main):004:0> URI.regexp =~ "git@github.com:bundler/bundler.git"
=> 4
irb(main):006:0> URI.parse("git@github.com:bundler/bundler.git")
URI::InvalidURIError: bad URI(is not URI?): git@github.com:bundler/bundler.git
	from /Users/frsyuki/.rbenv/versions/2.2.2/lib/ruby/2.2.0/uri/rfc3986_parser.rb:66:in `split'
	from /Users/frsyuki/.rbenv/versions/2.2.2/lib/ruby/2.2.0/uri/rfc3986_parser.rb:72:in `parse'
	from /Users/frsyuki/.rbenv/versions/2.2.2/lib/ruby/2.2.0/uri/common.rb:226:in `parse'
	from (irb):6
```